tutorials/integration.qmd

```diff
@@ -0,0 +1,198 @@
+---
+title: Integration
+---
+
+```{julia}
+#| echo: false
+#| code-fold: true
+#| output: false
+using Pkg;
+cd(@__DIR__)
+Pkg.activate("."); # for reproducibility use the local tutorial environment.
+Pkg.develop(PackageSpec(; path=(@__DIR__) * "/../"))
+using Markdown
+```
+
+This part of documentation covers integration of scalar functions defined on manifolds $f \colon \mathcal{M} \to \mathbb{R}$:
+
+```math
+\int_{\mathcal M} f(p) \mathrm{d}p
+```
+
+The basic concepts are derived from geometric measure theory.
+In principle, there are many ways in which a manifold can be equipped with a measure that can be later used to define an integral.
+One of the most popular ways is based on pushing the Lebesgue measure on a tangent space through the exponential map.
+Any other suitable atlas could be used, not just the one defined by normal coordinates, though each one requires different volume density corrections due to the Jacobian determinant of the pushforward.
+`Manifolds.jl` provides the function [`volume_density`](@ref) that calculates that quantity, denoted $\theta_p(X)$.
+See for example [^leBrigantPuechmorel2019], Definition 11, for a precise description using Jacobi fields.
+
+While many sources define volume density as a function of two points, `Manifolds.jl` decided to use the more general point-tangent vector formulation. The two-points variant can be implemented as
+```{julia}
+using Manifolds
+volume_density_two_points(M::AbstractManifold, p, q) = volume_density(M, p, log(M, p, q))
+```
+
+The simplest way to of integrating a function on a compact manifold is through a [📖 Monte Carlo integrator](https://en.wikipedia.org/wiki/Monte_Carlo_integration).
+A simple variant can be implemented as follows (assuming uniform distribution of `rand`):
+
+```{julia}
+using LinearAlgebra, Distributions, SpecialFunctions
+function simple_mc_integrate(M::AbstractManifold, f; N::Int = 1000)
+    V = manifold_volume(M)
+    sum = 0.0
+    q = rand(M)
+    for i in 1:N
+        sum += f(M, q)
+        rand!(M, q)
+    end
+    return V * sum/N
+end
+```
+
+We used the function [`manifold_volume`](@ref) to get the volume of the set over which integration is performed, as described in the linked Wikipedia article.
+
+## Distributions
+
+We will now try to verify that volume density correction correctly changes probability density of an exponential-wrapped normal distribution.
+`pdf_tangent_space` represents probability density of a normally distributed random variable $X_T$ in the tangent space $T_p \mathcal{M}$.
```

Comment:
Could you clarify here that `pdf_tangent_space` is a function that you plan to define (and do later)? same for `pdf_manifold`.

Reply:
Sure, done.